src/isaacs_mapping/src/process_aruco.py

```diff
@@ -1,95 +1,199 @@
+	"camera_dist_coeffs" : np.array([0.0, 0.0, 0.0, 0.0, 0.0]),
+	"camera_extrinsic" : [[1, 0, 0, 0.06],
+						[0, 1, 0, 0],
+						[0, 0, 1, 0],
+						[0, 0, 0, 1]],
+	"converted_point_cloud_node" : "converted_cloud" #name of the node we publish our converted point clouds to
+}
+
+class PointCloudCamToMarkerConverter:
+	def __init__(self, show_marker_UI = False, image_is_compressed = False, print_cam_location = False):
+		self.marker_size = SETTINGS["marker_size"]
+		self.aruco_dict = SETTINGS["aruco_dict"]
+		self.camera_intrinsic = SETTINGS["camera_intrinsic"]
+		self.camera_dist_coeffs = SETTINGS["camera_dist_coeffs"]
+		self.camera_extrinsic = SETTINGS["camera_extrinsic"]
+
+		self.pose_is_set = False
+		self.image_subsriber = None
+		self.pose_subsriber = None
+		self.pointcloud_subsriber = None
+		self.pointcloud_publisher = None
+		self.converted_point_cloud_node = SETTINGS["converted_point_cloud_node"]
+		self.camera_pose = None
+		self.zed2marker = [[1, 0, 0, 0], #the zed2marker conversion matrix
+								[0, 1, 0, 0], 
+								[0, 0, 1, 0], 
+								[0, 0, 0, 1]]
+		
+		self.calibrate_initial_pos = False #currently we assume camera is at its origin when detects the marker
+		self.print_camera_pos = print_cam_location
+		self.image_compressed = image_is_compressed
+		self.show_marker_UI = show_marker_UI
+
+
+	def start(self):		
+		rospy.init_node('tracking_data', anonymous=True)
+		#create the subscribers needed for detecting marker and set pose
+		if(self.image_compressed):
+			self.image_subsriber = rospy.Subscriber('/zed2/zed_node/rgb/image_rect_color/compressed', CompressedImage, self.process_image)
+		else:
+			self.image_subsriber = rospy.Subscriber('/zed2/zed_node/rgb/image_rect_color', Image, self.process_image)
+		self.pose_subsriber = rospy.Subscriber('/zed2/zed_node/pose', PoseStamped, self.update_camera_pose)
+		self.pointcloud_publisher = rospy.Publisher(self.converted_point_cloud_node, PointCloud2, queue_size=10)
+		rospy.spin()
+
+	"""Called when receiving an image message from Zed. Use the image to detect marker and set pose."""
+	def process_image(self, image_data):
+		if(self.pose_is_set):
+			return
+		#deserialize the image so it can be used for OpenCV ArUco marker detection
+		if(self.image_compressed):
+			image_serialized = np.fromstring(image_data.data, np.uint8)
+			image_deserialized = cv2.imdecode(image_serialized, cv2.IMREAD_COLOR)
+		else:
+			image_deserialized = np.frombuffer(image_data.data, dtype=np.uint8).reshape(image_data.height, image_data.width, -1)
+		
+		if(self.try_set_pose(image_deserialized)):
+			print("finish set pose")
+			self.pose_is_set = True
+			#after pose is set, update the subscriber and keep/create the ones needed for converting point clouds
+			self.image_subsriber.unregister()
+			if(not self.print_camera_pos):
+				self.pose_subsriber.unregister()
+			self.pointcloud_subsriber = rospy.Subscriber('/zed2/zed_node/mapping/fused_cloud', PointCloud2, self.convert_zed_pose)
```

Comment:
```suggestion
	""" Called when receiving an image message from ZED. Uses the image to detect the marker and set the pose."""
	def process_image(self, image_data):
		if self.pose_is_set:
			return
		# deserialize the image so it can be used for OpenCV ArUco marker detection
		if self.image_compressed:
			image_serialized = np.fromstring(image_data.data, np.uint8)
			image_deserialized = cv2.imdecode(image_serialized, cv2.IMREAD_COLOR)
		else:
			image_deserialized = np.frombuffer(image_data.data, dtype=np.uint8).reshape(image_data.height, image_data.width, -1)
		
		if self.try_set_pose(image_deserialized):
			print("finish set pose")
			self.pose_is_set = True
			# after the pose is set, update the subscriber and keep/create the ones needed for converting point clouds
			self.image_subsriber.unregister()
			if not self.print_camera_pos:
				self.pose_subsriber.unregister()
			self.pointcloud_subsriber = rospy.Subscriber('/zed2/zed_node/mapping/fused_cloud', PointCloud2, self.convert_zed_pose)
```